### ISSUE TITLE
SC problematic for Closed functionality: SC 2.1.2 No Keyboard Trap

### ISSUE THREAD

**Issue by pday1** [w3c/wcag2ict#272]
From closed functionality questionnaire:
https://www.w3.org/2002/09/wbs/55145/wcag2ict-sc-problematic-for-closed/results#xq26

Loïc Martínez Normand |  First, I agree with Phil about 2.4.7. In addition, 2.1.2 and 2.1.4 are also related to keyboard access. I suggest that we need them with notes similar to 2.1.1. Once we agree on the note for 2.1.1, the notes for 2.1.2, 2.1.4 and 2.4.7 could follow a similar approach.

Applying SC text included for reference
> 2.1.2 No Keyboard Trap: If keyboard focus can be moved to a component of the [[non-web document](https://w3c.github.io/wcag2ict/#document) or [software](https://w3c.github.io/wcag2ict/#software)] using a [keyboard interface](https://w3c.github.io/wcag2ict/#dfn-keyboard-interface), then focus can be moved away from that component using only a keyboard interface, and, if it requires more than unmodified arrow or tab keys or other standard exit methods, the user is advised of the method for moving focus away. (Level A)
> 
> NOTE 1
> Since any content that does not meet this success criterion can interfere with a user's ability to use the whole [[non-web document](https://w3c.github.io/wcag2ict/#document) or [software](https://w3c.github.io/wcag2ict/#software)], all content on the [non-web document or software] (whether it is used to meet other success criteria or not) must meet this success criterion.
> 
> NOTE 2
> Standard exit methods may vary by platform. For example, on many desktop platforms, the Escape key is a standard method for exiting.
>

**Comment by pday1** [comment]
Discussion from [survey results](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-SC-problematic-last-5/results#xq7) and [meeting minutes](https://www.w3.org/2024/01/05-wcag2ict-minutes) suggested that changes were needed.


Proposed update: 
[2.1.2 No Keyboard Trap](https://github.com/w3c/wcag2ict/issues/272#no-keyboard-trap) - 
This criterion applies when focus can be moved using a keyboard interface. Some software may accept input from a keyboard, keypad, or controller, yet not offer any mechanism for focus; for example, the keys are mapped directly to functions without moving focus between on-screen controls. In this case, there is no concept of focus, and therefore keyboard traps cannot exist and the intent of this success criterion is met.

**Comment by pday1** [comment]
For ease of review, here is content pulled from the latest Editor's draft to show the original "Applying SC to Non-Web..." alongside the proposed new content for "SC problematic for Closed...".

The latest editor's draft can be found at https://w3c.github.io/wcag2ict/
(Note this does not contain content that is in pull requests that are open and not yet approved)

**Content from Applying SC**





**Latest proposed content for SC problematic for Closed functionality section**

**Comment by maryjom** [comment]
See [Survey results on the last draft](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-SC-problematic-last-5/results#xq6). Friday sub-group drafted an updated proposal for the SC problematic for Closed functionality section. TF [discussed the updated draft for 2.1.2](https://www.w3.org/2024/01/18-wcag2ict-minutes#t03) on 18 January, but did not get to the SC Problematic bullet. We only decided that an additional note in the general guidance for 2.1.1 was not needed.

UPDATED proposal from the Friday sub-group ([developed on 5 January](https://www.w3.org/2024/01/05-wcag2ict-minutes)): 
NOTE 3: This criterion applies when focus can be moved using a keyboard interface. Some software may accept input from a keyboard, keypad, or controller, yet not offer any mechanism for focus; for example, the keys are mapped directly to functions without moving focus between on-screen controls. In this case, there is no concept of focus; therefore, keyboard traps cannot exist and this success criterion is satisfied.

**Comment by maryjom** [comment]
In his survey comment, Loïc said we should make this bullet similar to what exists for 2.1.1 Keyboard. The content for 2.1.1 was consensed on 12 October as follows:
>* [2.1.1 Keyboard](https://w3c.github.io/wcag2ict/#keyboard)—Requires operation via a keyboard interface which allows alternative input devices. When a product with closed functionality either does not have a standard keyboard or one cannot be connected, it would need an alternate way to access all functionality that does not require accurate pointing, path-based movements, or specific timings.

**Comment by maryjom** [comment]
@phil I updated to be consistent with other keyboard SC proposals I'm putting into the survey due 6 March. Just a change to the last 5 words.

**Comment by maryjom** [comment]
TF reached [resolution](https://www.w3.org/2024/03/07-wcag2ict-minutes#r01) on the "Applying SC..." Note 3 and new "SC Problematic for Closed Functionality" content for 2.1.2 in the 7 March meeting per the unanimous agreement in the [Survey results](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-closed-keyboard-scs/results#xq1).  Implemented in PR #318.

**Comment by maryjom** [comment]
The agreed content in PR #318 has been merged into the editor's draft, so this can be closed.
